code/modules/detective_work/swabs.dm

```diff
@@ -0,0 +1,186 @@
+/obj/item/clothing/clean_blood(radiation_clean = FALSE)
+	. = ..()
+	gunshot_residue = null
+
+/obj/item/forensics/swab
+	name = "sample collection kit"
+	desc = "Sterile cotton swab and test tube for collecting samples."
+	icon_state = "swab"
+	///currently in machine
+	var/dispenser = FALSE
+	///gunshot residue data
+	var/gsr = FALSE
+	///DNA samble data
+	var/list/dna
+	///boolean, used or not
+	var/used
+	///limits to one sample per item
+	var/inuse = FALSE
+
+/obj/item/forensics/swab/proc/is_used()
+	return used
+
+/obj/item/forensics/swab/pre_attack(atom/A, mob/living/user, params)
+	. = ..()
+	return A.attackby__legacy__attackchain(src, user, params)
```

Comment:
I don't really like this. Also, are you kinda mixing old and new attack chains here? I think the whole attack chain thing might need a revisit throughout this. I'd recommend poking someone more familiar with it in coding chat